library/mr_provisioner_netboot_switch.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import requests
+import time
+try:
+    from urllib.parse import urljoin
+except ImportError:
+    from urlparse import urljoin
+
+try:
+    from urllib import quote
+except ImportError:
+    from urllib.parse import quote
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: mr-provisioner-netboot-switch
+short_description: Manage the netboot_enabled switch in Mr Provisioner
+description:
+    Implemented:
+        - Switch the netboot flag off after a certain timeout
+    Not Implemented:
+        - Switch it on (as there is no use case aside from provisioning)
+options:
+    name:
+        description:
+            - Name of the machine whose flag needs to be turned off
+        required: true
+    timeout:
+        description:
+            - Timeout before turning the netboot flag off
+        required: false
+    path:
+        description: Local file path to preseed file.
+        required: true
+    url:
+        description: url to provisioner instance in the form of http://192.168.0.3:5000/
+        required: true
+    token:
+        description: Mr. Provisioner auth token
+        required: true
+author:
+    - Baptiste Gerondeau <baptiste.gerondeau@linaro.org>
+'''
+
+EXAMPLES = '''
+# Turns off test's netboot flag after 600 seconds
```

Comment:
What if the machine hasn't asked for the images during this period? This is prone to fail.